internal/controller/cloud-control/vpcpeering_aws_test.go

```diff
@@ -657,4 +650,231 @@ var _ = Describe("Feature: KCP VpcPeering", func() {
+				WithDetails(localKcpNetworkName, infra.KCP().Namespace(), remoteKcpNetworkName, infra.KCP().Namespace(), "", false, true).
+				Build()
+
+			Eventually(CreateObj).
+				WithArguments(infra.Ctx(), infra.KCP().Client(), kcpPeering,
+					WithName(kcpPeeringName),
+				).Should(Succeed())
+
+		})
+
+		By("And Given KCP VpcPeering have status.id set", func() {
+			Eventually(LoadAndCheck).
+				WithArguments(infra.Ctx(), infra.KCP().Client(), kcpPeering,
+					NewObjActions(),
+					HaveFinalizer(cloudcontrolv1beta1.FinalizerName),
+					HavingKcpVpcPeeringStatusIdNotEmpty(),
+				).Should(Succeed())
+		})
+
+		By("And Given AWS VpcPeeringConnections are active", func() {
+
+			// initiate remote vpc peering connection
+			awsMockRemote.InitiateVpcPeeringConnection(ptr.To(kcpPeering.Status.Id), ptr.To(localVpcId), ptr.To(remoteVpcId))
+
+			// change local vpc peering status to pending-acceptance (not necessary but leaving it for the clarity)
+			awsMockLocal.SetVpcPeeringConnectionStatusCode(ptr.To(localVpcId), ptr.To(remoteVpcId), ec2Types.VpcPeeringConnectionStateReasonCodePendingAcceptance)
+
+			// sets vpc peering connections active
+			awsMockLocal.SetVpcPeeringConnectionStatusCode(ptr.To(localVpcId), ptr.To(remoteVpcId), ec2Types.VpcPeeringConnectionStateReasonCodeActive)
+			awsMockRemote.SetVpcPeeringConnectionStatusCode(ptr.To(localVpcId), ptr.To(remoteVpcId), ec2Types.VpcPeeringConnectionStateReasonCodeActive)
+		})
+
+		By("And Given VpcPeering is Ready", func() {
+			Eventually(LoadAndCheck).
+				WithArguments(infra.Ctx(), infra.KCP().Client(), kcpPeering,
+					NewObjActions(),
+					HavingConditionTrue(cloudcontrolv1beta1.ConditionTypeReady)).
+				Should(Succeed())
+		})
+
+		By("And Then KCP VpcPeering has status.id equals to status.remoteId", func() {
+			Expect(kcpPeering.Status.RemoteId).To(Equal(kcpPeering.Status.Id))
+		})
+
+		By("And Then KCP VpcPeering has status.vpcId equals to existing AWS VPC id", func() {
+			Expect(kcpPeering.Status.VpcId).To(Equal(localVpcId))
+		})
+
+		By("And Then found local VpcPeeringConnection has AccepterVpcInfo.VpcId equals remote vpc id", func() {
+			localPeering, _ := awsMockLocal.DescribeVpcPeeringConnection(infra.Ctx(), kcpPeering.Status.Id)
+			Expect(*localPeering.AccepterVpcInfo.VpcId).To(Equal(remoteVpcId))
+		})
+
+		By("And Then there are no additional route entries added to local route tables", func() {
+			Expect(awsMockLocal.GetRouteCount(infra.Ctx(), localVpcId, kcpPeering.Status.Id, remoteVpcCidr)).
+				To(Equal(2))
+		})
+
+		By("And Then there are no additional route entries added to remote route tables", func() {
+			Expect(awsMockRemote.GetRouteCount(infra.Ctx(), remoteVpcId, kcpPeering.Status.RemoteId, localVpcCidr)).
+				To(Equal(2))
+		})
+
+		By("And Then remote VpcPeeringConnection is not deleted", func() {
+			remotePeering, _ := awsMockRemote.DescribeVpcPeeringConnection(infra.Ctx(), kcpPeering.Status.Id)
+			Expect(remotePeering).NotTo(BeNil())
+		})
+
```

Comment:
In my opinion, this whole section is not necessary for this scenario
```suggestion
```

Reply:
Correct! This is already covered in different scenario.